celery_singleton/singleton.py

```diff
@@ -126,6 +136,14 @@ def lock_and_run(self, lock, *args, task_id=None, **kwargs):
                 # Clear the lock if apply_async fails
                 self.unlock(lock)
                 raise
+    
+    def challenge_lock(self, lock, args, kwargs):
+        current_task_lock = self.generate_lock(self.name, args, kwargs)
```

Comment:
This variable is never used?

Reply:
Yes, this variable is now passed as an argument (lock) and should not be evaluated here anymore -> removing it